### PR TITLE
Aura: added tick delay wait mode

### PR DESCRIPTION
**Pull request by blockparole**

### CODE REVIEW

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java [hunk 1]
```diff
@@ -178,4 +190,8 @@ private boolean canEntityFeetBeSeen(Entity entityIn) {
         return mc.world.rayTraceBlocks(new Vec3d(mc.player.posX, mc.player.posY + mc.player.getEyeHeight(), mc.player.posZ), new Vec3d(entityIn.posX, entityIn.posY, entityIn.posZ), false, true, false) == null;
     }
 
+    private enum WaitMode {
+        KAMI, TICK
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: `KAMI` doesn't really explain what this mode does: consider `Delay`/`Wait`/... ?

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java [hunk 2]
```diff
@@ -23,20 +23,23 @@
...
-    private Setting<Boolean> wait = register(Settings.b("Wait", true));
-    private Setting<Boolean> walls = register(Settings.b("Walls", false));
-    private Setting<Boolean> switchTo32k = register(Settings.b("32k Switch", false));
+    private Setting<Boolean> attackPlayers = register(Settings.b("Players", true));
+    private Setting<Boolean> attackMobs = register(Settings.b("Mobs", false));
+    private Setting<Boolean> attackAnimals = register(Settings.b("Animals", false));
+    private Setting<Double> hitRange = register(Settings.d("Hit Range", 5.5d));
+    private Setting<Boolean> ignoreWalls = register(Settings.b("Ignore Walls", true));
+    private Setting<WaitMode> waitMode = register(Settings.e("Mode", WaitMode.KAMI));
+    private Setting<Double> waitTick = register(Settings.d("Tick Wait", 3.0));
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: Make this settings's visibility depend on `waitMode` being equal to `WaitMode.TICK` please

Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: not really sure how to implement this, can i change the visibility of a setting after registering it?

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 1 above)
Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: How do you like `Dynamic` and `Static` for wait modes?

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 2 above)
Comment (zeroeightysix) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: The settings builders allow this functionality:

https://github.com/zeroeightysix/KAMI/blob/312074ad5f988b4801f712b948a47a745e36e43b/src/main/java/me/zeroeightsix/kami/module/modules/render/Brightness.java#L18

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 1 above)
Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: changed [here](https://github.com/blockparole/KAMI/commit/fcdde15967ef2b544caa4f159415cb0b0c2b928a)

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java (hunk 2 above)
Comment (blockparole) [reply src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: changed [here](https://github.com/blockparole/KAMI/commit/67ff9e0496dbd9568a47ffb637c05761dd3f85ed)

File: src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java [hunk 3]
```diff
@@ -191,7 +191,7 @@ private boolean canEntityFeetBeSeen(Entity entityIn) {
     }
 
     private enum WaitMode {
-        KAMI, TICK
+        Dynamic, Static
```
Comment (zeroeightysix) [inline src/main/java/me/zeroeightsix/kami/module/modules/combat/Aura.java]: Style: enums are `UPPER_CASE`

**Comment by zeroeightysix** [comment]
Thanks!

**Pull request merged**